marge.py

```diff
@@ -1,53 +1,176 @@
+import xml.etree.ElementTree as ET
 from datetime import datetime
-from typing import List
-
-from model import SourceProvider
-
-
-def source_providers() -> List:
-
-    providers = [
-        "PANDORA",
-        "INTERNET_RADIO",
-        "OFF",
-        "LOCAL",
-        "AIRPLAY",
-        "CURRATED_RADIO",
-        "STORED_MUSIC",
-        "SLAVE_SOURCE",
-        "AUX",
-        "RECOMMENDED_INTERNET_RADIO",
-        "LOCAL_INTERNET_RADIO",
-        "GLOBAL_INTERNET_RADIO",
-        "HELLO",
-        "DEEZER",
-        "SPOTIFY",
-        "IHEART",
-        "SIRIUSXM",
-        "GOOGLE_PLAY_MUSIC",
-        "QQMUSIC",
-        "AMAZON",
-        "LOCAL_MUSIC",
-        "WBMX",
-        "SOUNDCLOUD",
-        "TIDAL",
-        "TUNEIN",
-        "QPLAY",
-        "JUKE",
-        "BBC",
-        "DARFM",
-        "7DIGITAL",
-        "SAAVN",
-        "RDIO",
-        "PHONE_MUSIC",
-        "ALEXA",
-        "RADIOPLAYER",
-        "RADIO.COM",
-        "RADIO_COM",
-        "SIRIUSXM_EVEREST",
-    ]
+
+from config import Settings
+from model import ConfiguredSource, Preset, SourceProvider
+
+PROVIDERS = [
+    "PANDORA",
+    "INTERNET_RADIO",
+    "OFF",
+    "LOCAL",
+    "AIRPLAY",
+    "CURRATED_RADIO",
+    "STORED_MUSIC",
+    "SLAVE_SOURCE",
+    "AUX",
+    "RECOMMENDED_INTERNET_RADIO",
+    "LOCAL_INTERNET_RADIO",
+    "GLOBAL_INTERNET_RADIO",
+    "HELLO",
+    "DEEZER",
+    "SPOTIFY",
+    "IHEART",
+    "SIRIUSXM",
+    "GOOGLE_PLAY_MUSIC",
+    "QQMUSIC",
+    "AMAZON",
+    "LOCAL_MUSIC",
+    "WBMX",
+    "SOUNDCLOUD",
+    "TIDAL",
+    "TUNEIN",
+    "QPLAY",
+    "JUKE",
+    "BBC",
+    "DARFM",
+    "7DIGITAL",
+    "SAAVN",
+    "RDIO",
+    "PHONE_MUSIC",
+    "ALEXA",
+    "RADIOPLAYER",
+    "RADIO.COM",
+    "RADIO_COM",
+    "SIRIUSXM_EVEREST",
+]
+
+
+def account_device_dir(settings: Settings, account: str, device: str) -> str:
+    return settings.data_dir + "/" + account + "/" + device + "/"
+
+
+def source_providers() -> list[SourceProvider]:
     datestr = "2012-09-19T12:43:00.000+00:00"
     return [
         SourceProvider(id=i[0], created_on=datestr, name=i[1], updated_on=datestr)
-        for i in enumerate(providers, start=1)
+        for i in enumerate(PROVIDERS, start=1)
     ]
+
+
+def configured_sources(
+    settings: Settings, account: str, device: str
+) -> list[ConfiguredSource]:
+    sources_tree = ET.parse(
+        account_device_dir(settings, account, device) + "Sources.xml"
```

Comment:
in general making format strings is more pythonic than string concatenation if you don't have a reason to choose one over the other, but in this case it's a path so for OS-compatibility:

```
        path(account_device_dir(settings, account, device), "Sources.xml")
```